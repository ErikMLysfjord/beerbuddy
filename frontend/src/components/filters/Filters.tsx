import { Button, Checkbox, Collapse, Divider, Tooltip } from "antd";
import { Slider, SliderProps, alpha, styled } from "@mui/material";
import { useContext, useId, useRef } from "react";
import { FilterContext } from "../../context/FilterContext";
import {
  QuestionCircleOutlined,
  RollbackOutlined,
  RightOutlined,
} from "@ant-design/icons";
import { default as CSSstyles } from "./Filters.module.css";

// Styles of beer that can be filtered on
const beerStyles = [
  "American IPA",
  "American Pale Ale (APA)",
  "American Amber / Red Ale",
  "American Blonde Ale",
  "American Double / Imperial IPA",
  "American Pale Wheat Ale",
  "American Brown Ale",
  "American Porter",
  "Saison / Farmhouse Ale",
  "Witbier",
  "Fruit / Vegetable Beer",
  "KÃ¶lsch",
  "Hefeweizen",
  "American Pale Lager",
  "American Stout",
  "Other",
];

// Styled slider component from MUI
const StyledSlider = styled(Slider)<SliderProps>(() => ({
  color: "#FFCC48",
  "& .MuiSlider-thumb": {
    "&:hover, &.Mui-focusVisible": {
      boxShadow: `0px 0px 0px 8px ${alpha("#FFCC48", 0.16)}`,
    },
    "&.Mui-active": {
      boxShadow: `0px 0px 0px 14px ${alpha("#FFCC48", 0.16)}`,
    },
  },
}));

interface FiltersProps {
  fetchMore: (reset?: boolean) => Promise<void>;
  apply?: () => void;
}
/**
 * Filters component that contains all filters.
 * @param fetchMore - function that is called when the apply filters button is clicked
 * @param apply - function that is called to hide the filters when the apply filters button is clicked
 * @returns a Filters component
 */
const Filters = ({ fetchMore, apply }: FiltersProps) => {
  const {
    IBU,
    setIBU,
    ABV,
    setABV,
    setStyles,
    styles,
    searchString,
    setSearchString,
  } = useContext(FilterContext);
  const fetchMoreRef = useRef(fetchMore);
  fetchMoreRef.current = fetchMore;
  const headingID = useId();

  /**
   * Function for resetting filters and fetching more beers.
   */
  const resetFilters = () => {
    /* Only reset filters and fetch beers if there are active filters */
    if (
      beerStyles.length > 0 ||
      ABV[0] !== 0 ||
      ABV[1] !== 40 ||
      IBU[0] !== 0 ||
      IBU[1] !== 140 ||
      searchString !== ""
    ) {
      setABV([0, 40]);
      setIBU([0, 140]);
      setStyles([]);
      setSearchString("");
      fetchMoreRef.current(true);
    }
  };

  return (
    <section aria-labelledby={headingID}>
      <Divider />
      <h2 id={headingID}>Filters</h2>

      <Collapse
        aria-label="Beer styles"
        expandIcon={({ isActive }) => (
          <RightOutlined rotate={isActive ? 90 : 0} aria-hidden />
        )}
        className={CSSstyles.collapse}
        items={[
          {
            key: "styles-checkbox-key",
            label: (
              <span className={CSSstyles.collapseSpan}>
                Beer styles
                <Tooltip title={"Filter on certain styles of beer."}>
                  <QuestionCircleOutlined aria-hidden />
                </Tooltip>
              </span>
            ),
            children: (
              <Checkbox.Group
                options={beerStyles}
                name="styles"
                onChange={(checkedValues) =>
                  setStyles(checkedValues.map((value) => value.toString()))
                }
                value={styles}
              />
            ),
          },
        ]}
      />
      <Collapse
        expandIcon={({ isActive }) => (
          <RightOutlined rotate={isActive ? 90 : 0} aria-hidden />
        )}
        className={CSSstyles.collapse}
        items={[
          {
            key: "IBU-range-key",
            label: (
              <span className={CSSstyles.collapseSpan}>
                IBU
                <Tooltip title={"Filter on the bitterness of the beer."}>
                  <QuestionCircleOutlined aria-hidden />
                </Tooltip>
              </span>
            ),
            children: (
              <StyledSlider
                getAriaLabel={() => "IBU range"}
                value={IBU}
                onChange={(e: Event, newValue: number | number[]) => {
                  e.preventDefault();
                  setIBU(newValue as number[]);
                }}
                max={140}
                valueLabelDisplay="auto"
                getAriaValueText={() =>
                  `Min value is ${IBU[0]}, max is ${IBU[1]}`
                }
              />
            ),
          },
        ]}
      />
      <Collapse
        className={CSSstyles.collapse}
        expandIcon={({ isActive }) => (
          <RightOutlined rotate={isActive ? 90 : 0} aria-hidden />
        )}
        items={[
          {
            key: "ABV-range-key",
            label: (
              <span className={CSSstyles.collapseSpan}>
                ABV
                <Tooltip
                  title={"Filter on the alcohol percentage of the beer."}
                >
                  <QuestionCircleOutlined aria-hidden />
                </Tooltip>
              </span>
            ),
            children: (
              <StyledSlider
                getAriaLabel={() => "ABV range"}
                value={ABV}
                onChange={(e: Event, newValue: number | number[]) => {
                  e.preventDefault();
                  setABV(newValue as number[]);
                }}
                max={40}
                valueLabelDisplay="auto"
                getAriaValueText={() =>
                  `Min value is ${ABV[0]}, max is ${ABV[1]}`
                }
              />
            ),
          },
        ]}
      />

      <Button
        type="primary"
        className={CSSstyles.buttons}
        onClick={(e) => {
          e.preventDefault();
          if (apply) apply();
          fetchMore(true);
        }}
      >
        Apply Filters
      </Button>
      <Button
        type="default"
        className={CSSstyles.buttons}
        onClick={resetFilters}
        icon={<RollbackOutlined />}
      >
        Reset Filters
      </Button>
    </section>
  );
};
export default Filters;
