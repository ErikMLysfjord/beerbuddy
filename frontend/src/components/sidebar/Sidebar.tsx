import Logo from "../logo/Logo";
import styles from "./Sidebar.module.css";

interface SidebarProps {
  children: React.ReactNode;
}

/**
 * Sidebar component that contains the BeerBuddy logo
 * and a main section for the filter components.
 * It also contains a button that applies the filter settings
 * to the search result.
 * @param children - filter components
 * @param fetchMore - function that is called when the apply filters button is clicked
 * @returns a Sidebar component
 */
const Sidebar = ({ children }: SidebarProps) => {
  return (
    <section className={styles.container}>
      <div>
        <Logo />
        <section>{children}</section>
      </div>
    </section>
  );
};
export default Sidebar;
