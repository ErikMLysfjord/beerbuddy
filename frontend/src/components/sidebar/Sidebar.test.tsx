import { describe, it, expect, vi } from "vitest";
import { fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Sidebar from "./Sidebar";
import { axe } from "jest-axe";
import Filters from "../filters/Filters";

const mockFetchMore = vi.fn(async () => await Promise.resolve());

const Template = () => (
  <Sidebar>
    <Filters fetchMore={mockFetchMore} />
    <p>Test</p>
  </Sidebar>
);

describe("Sidebar", () => {
  it("is accessible", async () => {
    const { container } = render(<Template />);
    expect(await axe(container)).toHaveNoViolations();
  });

  it("renders correctly", () => {
    const { container } = render(<Template />);
    expect(container).toMatchSnapshot();
  });

  it("renders with correct children", () => {
    const { getByText } = render(<Template />);
    expect(getByText("Test")).toBeInTheDocument();
  });

  it("renders with correct button", () => {
    const { getByText } = render(<Template />);
    const applyButton = getByText("Apply Filters");
    expect(applyButton).toBeInTheDocument();
    expect(applyButton).toHaveTextContent("Apply Filters");
  });

  it("calls fetchMore when button is clicked", () => {
    const { getByText } = render(<Template />);
    const applyButton = getByText("Apply Filters");
    fireEvent.click(applyButton);
    expect(mockFetchMore).toHaveBeenCalled();
  });
});
