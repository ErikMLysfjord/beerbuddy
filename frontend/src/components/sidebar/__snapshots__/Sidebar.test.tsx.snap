// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sidebar > renders correctly 1`] = `
<div>
  <section
    aria-label="Sidebar"
    class="_container_26077f"
  >
    <div>
      <a
        class="_wrapper_4f3c83"
        href="/project2"
      >
        <img
          alt="BeerBuddy logo"
          class="_icon_4f3c83"
          src="/project2/teku.svg"
        />
        <h1
          class="_heading_4f3c83"
        >
          BeerBuddy
        </h1>
      </a>
      <section
        aria-label="Filters"
      >
        <div
          class="ant-divider css-dev-only-do-not-override-pr0fja ant-divider-horizontal"
          role="separator"
        />
        <div
          class="_headingWrapper_3ea492"
        >
          <h2>
            Style
          </h2>
          <span
            aria-label="question-circle"
            class="anticon anticon-question-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="question-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
              <path
                d="M623.6 316.7C593.6 290.4 554 276 512 276s-81.6 14.5-111.6 40.7C369.2 344 352 380.7 352 420v7.6c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V420c0-44.1 43.1-80 96-80s96 35.9 96 80c0 31.1-22 59.6-56.1 72.7-21.2 8.1-39.2 22.3-52.1 40.9-13.1 19-19.9 41.8-19.9 64.9V620c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8v-22.7a48.3 48.3 0 0130.9-44.8c59-22.7 97.1-74.7 97.1-132.5.1-39.3-17.1-76-48.3-103.3zM472 732a40 40 0 1080 0 40 40 0 10-80 0z"
              />
            </svg>
          </span>
        </div>
        <div
          class="ant-checkbox-group css-dev-only-do-not-override-pr0fja"
        >
          <label
            class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-pr0fja"
          >
            <span
              class="ant-checkbox ant-wave-target css-dev-only-do-not-override-pr0fja"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
                value="American IPA"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              American IPA
            </span>
          </label>
          <label
            class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-pr0fja"
          >
            <span
              class="ant-checkbox ant-wave-target css-dev-only-do-not-override-pr0fja"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
                value="American Pale Ale (APA)"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              American Pale Ale (APA)
            </span>
          </label>
          <label
            class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-pr0fja"
          >
            <span
              class="ant-checkbox ant-wave-target css-dev-only-do-not-override-pr0fja"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
                value="American Amber / Red Ale"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              American Amber / Red Ale
            </span>
          </label>
          <label
            class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-pr0fja"
          >
            <span
              class="ant-checkbox ant-wave-target css-dev-only-do-not-override-pr0fja"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
                value="American Blonde Ale"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              American Blonde Ale
            </span>
          </label>
          <label
            class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-pr0fja"
          >
            <span
              class="ant-checkbox ant-wave-target css-dev-only-do-not-override-pr0fja"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
                value="American Double / Imperial IPA"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              American Double / Imperial IPA
            </span>
          </label>
          <label
            class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-pr0fja"
          >
            <span
              class="ant-checkbox ant-wave-target css-dev-only-do-not-override-pr0fja"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
                value="American Pale Wheat Ale"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              American Pale Wheat Ale
            </span>
          </label>
          <label
            class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-pr0fja"
          >
            <span
              class="ant-checkbox ant-wave-target css-dev-only-do-not-override-pr0fja"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
                value="American Brown Ale"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              American Brown Ale
            </span>
          </label>
          <label
            class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-pr0fja"
          >
            <span
              class="ant-checkbox ant-wave-target css-dev-only-do-not-override-pr0fja"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
                value="American Porter"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              American Porter
            </span>
          </label>
          <label
            class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-pr0fja"
          >
            <span
              class="ant-checkbox ant-wave-target css-dev-only-do-not-override-pr0fja"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
                value="Saison / Farmhouse Ale"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              Saison / Farmhouse Ale
            </span>
          </label>
          <label
            class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-pr0fja"
          >
            <span
              class="ant-checkbox ant-wave-target css-dev-only-do-not-override-pr0fja"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
                value="Witbier"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              Witbier
            </span>
          </label>
          <label
            class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-pr0fja"
          >
            <span
              class="ant-checkbox ant-wave-target css-dev-only-do-not-override-pr0fja"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
                value="Fruit / Vegetable Beer"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              Fruit / Vegetable Beer
            </span>
          </label>
          <label
            class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-pr0fja"
          >
            <span
              class="ant-checkbox ant-wave-target css-dev-only-do-not-override-pr0fja"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
                value="Kölsch"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              Kölsch
            </span>
          </label>
          <label
            class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-pr0fja"
          >
            <span
              class="ant-checkbox ant-wave-target css-dev-only-do-not-override-pr0fja"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
                value="Hefeweizen"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              Hefeweizen
            </span>
          </label>
          <label
            class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-pr0fja"
          >
            <span
              class="ant-checkbox ant-wave-target css-dev-only-do-not-override-pr0fja"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
                value="American Pale Lager"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              American Pale Lager
            </span>
          </label>
          <label
            class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-pr0fja"
          >
            <span
              class="ant-checkbox ant-wave-target css-dev-only-do-not-override-pr0fja"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
                value="American Stout"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              American Stout
            </span>
          </label>
          <label
            class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-pr0fja"
          >
            <span
              class="ant-checkbox ant-wave-target css-dev-only-do-not-override-pr0fja"
            >
              <input
                class="ant-checkbox-input"
                type="checkbox"
                value="Other"
              />
              <span
                class="ant-checkbox-inner"
              />
            </span>
            <span>
              Other
            </span>
          </label>
        </div>
        <div
          class="ant-divider css-dev-only-do-not-override-pr0fja ant-divider-horizontal"
          role="separator"
        />
        <div
          class="_headingWrapper_3ea492"
        >
          <h2>
            IBU
          </h2>
          <span
            aria-label="question-circle"
            class="anticon anticon-question-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="question-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
              <path
                d="M623.6 316.7C593.6 290.4 554 276 512 276s-81.6 14.5-111.6 40.7C369.2 344 352 380.7 352 420v7.6c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V420c0-44.1 43.1-80 96-80s96 35.9 96 80c0 31.1-22 59.6-56.1 72.7-21.2 8.1-39.2 22.3-52.1 40.9-13.1 19-19.9 41.8-19.9 64.9V620c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8v-22.7a48.3 48.3 0 0130.9-44.8c59-22.7 97.1-74.7 97.1-132.5.1-39.3-17.1-76-48.3-103.3zM472 732a40 40 0 1080 0 40 40 0 10-80 0z"
              />
            </svg>
          </span>
        </div>
        <span
          class="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-9h41w3-MuiSlider-root"
        >
          <span
            class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
          />
          <span
            class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
            style="left: 0%; width: 100%;"
          />
          <span
            class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-eg0mwd-MuiSlider-thumb"
            data-index="0"
            style="left: 0%;"
          >
            <input
              aria-label="IBU range"
              aria-orientation="horizontal"
              aria-valuemax="140"
              aria-valuemin="0"
              aria-valuenow="0"
              aria-valuetext="Min value is 0, max is 140"
              data-index="0"
              max="140"
              min="0"
              step="1"
              style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
              type="range"
              value="0"
            />
            <span
              aria-hidden="true"
              class="MuiSlider-valueLabel css-nnid7-MuiSlider-valueLabel"
            >
              <span
                class="MuiSlider-valueLabelCircle"
              >
                <span
                  class="MuiSlider-valueLabelLabel"
                >
                  0
                </span>
              </span>
            </span>
          </span>
          <span
            class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-eg0mwd-MuiSlider-thumb"
            data-index="1"
            style="left: 100%;"
          >
            <input
              aria-label="IBU range"
              aria-orientation="horizontal"
              aria-valuemax="140"
              aria-valuemin="0"
              aria-valuenow="140"
              aria-valuetext="Min value is 0, max is 140"
              data-index="1"
              max="140"
              min="0"
              step="1"
              style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
              type="range"
              value="140"
            />
            <span
              aria-hidden="true"
              class="MuiSlider-valueLabel css-nnid7-MuiSlider-valueLabel"
            >
              <span
                class="MuiSlider-valueLabelCircle"
              >
                <span
                  class="MuiSlider-valueLabelLabel"
                >
                  140
                </span>
              </span>
            </span>
          </span>
        </span>
        <div
          class="ant-divider css-dev-only-do-not-override-pr0fja ant-divider-horizontal"
          role="separator"
        />
        <div
          class="_headingWrapper_3ea492"
        >
          <h2>
            Alcohol
          </h2>
          <span
            aria-label="question-circle"
            class="anticon anticon-question-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="question-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
              <path
                d="M623.6 316.7C593.6 290.4 554 276 512 276s-81.6 14.5-111.6 40.7C369.2 344 352 380.7 352 420v7.6c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V420c0-44.1 43.1-80 96-80s96 35.9 96 80c0 31.1-22 59.6-56.1 72.7-21.2 8.1-39.2 22.3-52.1 40.9-13.1 19-19.9 41.8-19.9 64.9V620c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8v-22.7a48.3 48.3 0 0130.9-44.8c59-22.7 97.1-74.7 97.1-132.5.1-39.3-17.1-76-48.3-103.3zM472 732a40 40 0 1080 0 40 40 0 10-80 0z"
              />
            </svg>
          </span>
        </div>
        <span
          class="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-9h41w3-MuiSlider-root"
        >
          <span
            class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
          />
          <span
            class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
            style="left: 0%; width: 100%;"
          />
          <span
            class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-eg0mwd-MuiSlider-thumb"
            data-index="0"
            style="left: 0%;"
          >
            <input
              aria-label="ABV range"
              aria-orientation="horizontal"
              aria-valuemax="40"
              aria-valuemin="0"
              aria-valuenow="0"
              aria-valuetext="Min value is 0, max is 40"
              data-index="0"
              max="40"
              min="0"
              step="1"
              style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
              type="range"
              value="0"
            />
            <span
              aria-hidden="true"
              class="MuiSlider-valueLabel css-nnid7-MuiSlider-valueLabel"
            >
              <span
                class="MuiSlider-valueLabelCircle"
              >
                <span
                  class="MuiSlider-valueLabelLabel"
                >
                  0
                </span>
              </span>
            </span>
          </span>
          <span
            class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-eg0mwd-MuiSlider-thumb"
            data-index="1"
            style="left: 100%;"
          >
            <input
              aria-label="ABV range"
              aria-orientation="horizontal"
              aria-valuemax="40"
              aria-valuemin="0"
              aria-valuenow="40"
              aria-valuetext="Min value is 0, max is 40"
              data-index="1"
              max="40"
              min="0"
              step="1"
              style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
              type="range"
              value="40"
            />
            <span
              aria-hidden="true"
              class="MuiSlider-valueLabel css-nnid7-MuiSlider-valueLabel"
            >
              <span
                class="MuiSlider-valueLabelCircle"
              >
                <span
                  class="MuiSlider-valueLabelLabel"
                >
                  40
                </span>
              </span>
            </span>
          </span>
        </span>
        <button
          class="ant-btn css-dev-only-do-not-override-pr0fja ant-btn-primary"
          style="width: 100%; margin-top: 1rem;"
          type="button"
        >
          <span>
            Apply Filters
          </span>
        </button>
        <button
          class="ant-btn css-dev-only-do-not-override-pr0fja ant-btn-default"
          style="width: 100%; margin-top: 1rem;"
          type="button"
        >
          <span
            class="ant-btn-icon"
          >
            <span
              aria-label="rollback"
              class="anticon anticon-rollback"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="rollback"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M793 242H366v-74c0-6.7-7.7-10.4-12.9-6.3l-142 112a8 8 0 000 12.6l142 112c5.2 4.1 12.9.4 12.9-6.3v-74h415v470H175c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h618c35.3 0 64-28.7 64-64V306c0-35.3-28.7-64-64-64z"
                />
              </svg>
            </span>
          </span>
          <span>
            Reset Filters
          </span>
        </button>
        <p>
          Test
        </p>
      </section>
    </div>
  </section>
</div>
`;
